classes/session/SessionManager.php

```diff
@@ -317,6 +223,171 @@ public static function hasSession(): bool
         // If the session isn't disabled and a cookie is present or a session was started in the current request
         return !SessionManager::isDisabled() && (isset($_COOKIE[Config::getVar('general', 'session_cookie_name')]) || !!session_id());
     }
+
+    private function configure(): void
+    {
+        $domain = $this->request->getServerHost(includePort: false);
+
+        // Configure PHP session parameters
+        ini_set('session.name', Config::getVar('general', 'session_cookie_name'));
+        ini_set('session.cookie_lifetime', 0);
+        ini_set('session.cookie_path', Config::getVar('general', 'session_cookie_path', $this->request->getBasePath() . '/'));
+        ini_set('session.cookie_domain', $domain);
+        ini_set('session.cookie_httponly', 1);
+        ini_set('session.cookie_samesite', Config::getVar('general', 'same_site', 'Lax'));
+        ini_set('session.cookie_secure', Config::getVar('security', 'force_ssl'));
+        ini_set('session.use_trans_sid', 0);
+        ini_set('session.serialize_handler', 'php');
+        ini_set('session.use_cookies', 1);
+        ini_set('session.gc_probability', 1);
+        ini_set('session.gc_maxlifetime', 60 * 60);
+        ini_set('session.cache_limiter', 'none');
+
+        session_set_save_handler($this, true);
+    }
+
+    /**
+     * Starts the session
+     *
+     * In case there are many session cookies, it attempts to find the best one and clear the remaining, considering the issues below:
+     * Empty domains: Old applications (e.g. OJS 2.x) didn't store the domain, therefore users revising the application after a migration might be affected, we'll drop it.
+     * Subdomains mixed with parent domains: Users visiting "journal.sfu.ca", then "www.journal.sfu.ca" might end up with N+1 session cookies, this is problematic, we'll try to drop the excess.
+     * Domain cookies belonging to different paths or expired sessions: They will be kept (until the user clears his cookies) and probably trigger the extra checks.
+     */
+    private function start(): void
+    {
+        $sessionIds = collect($this->getSessionIds());
+        // Standard flow with a single session ID
+        if ($sessionIds->count() < 2) {
+            session_start();
+            return;
+        }
+
+        $requestDomain = Stringy::create($this->request->getServerHost(includePort: false));
+        /** @var \Illuminate\Support\Collection<Session> */
+        $sessions = $sessionIds
+            ->map(fn (string $sessionId) => $this->sessionDao->getSession($sessionId))
+            // Only sessions with valid domains (empty domains are also accepted)
+            ->filter(fn (?Session $session) => $session && $requestDomain->endsWith($session->getDomain(), false));
+
+        /** @var ?Session */
+        $bestSession = $sessions->reduce(function (?Session $best, Session $current): Session {
```

Comment:
Here's the fix for the multi-session issue, it just tries to find the best possible and will probably stick to it.